src/kernels/MIOpenSubTensorOpWithTransformKernel.cl

```diff
@@ -100,13 +100,42 @@
 #define WORK_STRIDE_1 (WORK_LENGTH_2 * WORK_STRIDE_2)
 #define WORK_STRIDE_0 (WORK_LENGTH_1 * WORK_STRIDE_1)
 
-#ifndef SUBTENSOR_OP_WITH_SCALAR
-#define SUBTENSOR_OP_WITH_SCALAR BREAK_COMPILE_INTENTIONALLY
+#ifndef MIOPEN_BETA_IS_ZERO
+#error "MIOPEN_BETA_IS_ZERO must be defined"
+#endif
+#ifndef MIOPEN_ALPHA_IS_ONE
+#error "MIOPEN_ALPHA_IS_ONE must be defined"
 #endif
 
-#define SUBTENSOR_OP_WITH_SCALAR_SET(t, a) (t = a)
-#define SUBTENSOR_OP_WITH_SCALAR_MULTIPLY(t, a) (t *= a)
-#define SUBTENSOR_OP_WITH_SCALAR_MAD(tb, b, ta, a) (tb = mad(ta, a, tb * b))
+#if MIOPEN_BETA_IS_ZERO && MIOPEN_ALPHA_IS_ONE
+#define SUBTENSOR_OP_WITH_ALPHA_BETA(dst, src) \
+    do                                         \
+    {                                          \
+        (dst) = (src);                         \
+        (void)beta;                            \
+        (void)alpha;                           \
+    } while(0)
+#elif MIOPEN_BETA_IS_ZERO
+#define SUBTENSOR_OP_WITH_ALPHA_BETA(dst, src) \
+    do                                         \
+    {                                          \
+        (dst) = (src)*alpha;                   \
+        (void)beta;                            \
+    } while(0)
+#elif MIOPEN_ALPHA_IS_ONE
+#define SUBTENSOR_OP_WITH_ALPHA_BETA(dst, src) \
+    do                                         \
+    {                                          \
+        (dst) = (src) + (dst)*beta;            \
```

Comment:
@junliume @CAHEK7 Maybe
```suggestion
        (dst) = mad((dst), beta, (src));       \
```
what do you think? This should produce one inst instead of two.